qiskit/pulse/compiler/passes/flatten.py

```diff
@@ -0,0 +1,95 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A flattening pass for Qiskit PulseIR compilation."""
+
+from __future__ import annotations
+
+from qiskit.pulse.compiler.basepasses import TransformationPass
+from qiskit.pulse.ir import SequenceIR
+from qiskit.pulse.exceptions import PulseCompilerError
+
+
+class Flatten(TransformationPass):
+    """Flatten ``SequenceIR`` object.
+
+    The flattening process includes breaking up nested IRs until only instructions remain.
+    After flattening the object will contain all instructions, timing information, and the
+    complete sequence graph. However, the alignment of nested IRs will be lost. Because alignment
+    information is essential for scheduling, flattening an unscheduled IR is not allowed.
+    One should apply :class:`~qiskit.pulse.compiler.passes.SetSchedule` first.
+    """
+
+    def __init__(self):
+        """Create new Flatten pass"""
+        super().__init__(target=None)
+
+    def run(
+        self,
+        passmanager_ir: SequenceIR,
+    ) -> SequenceIR:
+        """Run the pass."""
+
+        self._flatten(passmanager_ir)
+        return passmanager_ir
+
+    # pylint: disable=cell-var-from-loop
+    def _flatten(self, prog: SequenceIR) -> SequenceIR:
```

Comment:
```suggestion
    def _flatten_recursion(self, prog: SequenceIR) -> SequenceIR:
```
To align with your convention.